File: python/p002.py
import math

def compute(LIMIT):
	PHI = (1 + math.sqrt(5)) / 2
	sum = 0
	term = 3
	while sum < (LIMIT / 3):
		sum += int((math.pow(PHI, term) - math.pow(1 - PHI, term)) / math.sqrt(5))
		term += 3
	return sum

if __name__ == "__main__":
	print(compute(4000000))

File: c/p002.c
#include "math.h"
#include "stdio.h"

int compute(const int LIMIT) {
	const double phi = (1 + sqrt(5)) / 2;
	int sum = 0;
	int term = 3;
	while (sum < (LIMIT / 3)) {
		sum += (pow(phi, term) - pow((1 - phi), term)) / sqrt(5);
		term += 3;
	}
	return sum;
}

int main() {
	printf("%i\n", compute(4000000));
	return 0;
}
